libs/coroutines/include/hpx/coroutines/thread_enums.hpp

```diff
@@ -93,6 +93,10 @@ namespace hpx { namespace threads {
             priority queue and will be executed before normal/low priority
             tasks are taken (some schedulers modify the behavior slightly and
             the documentation for those should be consulted). */
+        thread_priority_bound = 6,          /*!< Task goes onto a special
+            high priority queue and will never be stolen by another thread
+            after initial assignment. This should be used for thread placement
+            tasks such as OpenMP type for loops. */
```

Comment:
Should this be a hint instead?

Reply:
Ideally we would have a better mechanism for passing this info, but for now I decided to go with a thread priority because the bound queues are considered high-priority as well as steal-free. As per the previous comment, I believe we need a root-and-branch cleanup of the design of some of this stuff and would prefer to get a working redesign of the scheduler in now and then make other changes later. 
The problem with a 'hint' is that it is not mandatory - ideally we should have `hints` and `orders` - which must be obeyed. Also the other schedulers cannot currently handle this and I didn't want to start redesigning all of them to accommodate a new  feature.
option 1) get this PR in - option 2) keep adding stuff to it for another year and never get it done. I went with option 1) in the hope that we can get the scheduler improvements in now and do some further design later.

Reply:
It is just a hint now because the other schedulers don't support it. I think it should be independent of priorities because you might want normal and low priority bound tasks. Whether it should be a hint or something entirely else though I don't know...

Reply:
yes. we would like low priority bound tasks, and high, but adding support for those will require more work and will delay the PR considerably ...

Reply:
I'll see how much I need to change to use a hint instead